packages/edit-site/src/components/style-book/constants.ts

```diff
@@ -239,6 +239,11 @@ export const STYLE_BOOK_IFRAME_STYLES = `
 		text-transform: uppercase;
 	}
 
+	.edit-site-style-book__example-subtitle {
+		font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
+		font-size: 9px;
+	}
```

Comment:
It is very weird that we're rendering the titles and subtitles... within an iframe. Do we have any context on this decision? I think it would have been a lot simpler to avoid an iframe for style book and only use iframes for the examples (maybe even use `BlockPreview` rather than `BlockList` and `Provider`) 

For clarity, I'm not asking of any change in this particular PR but I wonder if this was considered and if we didn't, we should probably consider as part of our efforts to improve style book. cc @tellthemachines @ramonjd @aaronrobertshaw

Reply:
Speaking of the font size, it's a bit small. The later mockups have two heading sizes: the main category (16px), and the subheading size (13px):
![Screenshot 2024-12-05 at 10 58 41](https://github.com/user-attachments/assets/b0fae1fe-e95f-4064-aa70-ea98893d7b0f)

Potentially this can be separate since there's a lot of separate work going on with headings, but there's a [spec here](https://www.figma.com/design/BAssQhQUVLPE1ld42m2Dn6/Global-Styles-%2F-Saving-Style-Variations-%2F-Text-Wrap?node-id=388-22717&t=XzcSrPw5i1IFIW1q-1).

Reply:
> Do we have any context on this decision? I think it would have been a lot simpler to avoid an iframe for style book and only use iframes for the examples

It's an abstraction of noisysock's [original implementation](https://github.com/WordPress/gutenberg/pull/45960/files#diff-d3b3586c65a2fff4adb274402a5c915ee5c819655af7b75819e9e101c4acb5aeR32). 

🤔 If I recall correctly, at the time it was to simulate an "editor instance" within the editor, but isolated so as to have more control over window resizing, styles and so on. It proved useful for revisions where discrete global styles configs could be loaded into the iframe without affecting the main canvas.

As is with many features, things just grew from there. 

> I think it would have been a lot simpler to avoid an iframe for style book and only use iframes for the examples (maybe even use BlockPreview rather than BlockList and Provider)
> I wonder if this was considered and if we didn't, we should probably consider as part of our efforts to improve style book

I agree the setup is ripe for refactoring. 

I tried a few variations while researching for https://github.com/WordPress/gutenberg/issues/62216. I think it's worth reigniting, so maybe that issue could be extended/repurposed.

It will be a whole lot simpler if Global styles (including revisions and style book) finds a new home in the left-hand "view" site editor side bar. 

There have been noises to that effect, but I doubt it will happen soon. 

By "simpler" I mean that the component wouldn't have to live in the editor canvas and do all the iframe/slot faffery.

Reply:
> By "simpler" I mean that the component wouldn't have to live in the editor canvas and do all the iframe/slot faffery.

I feel like refactoring the stylebook to use multiple small "Block Previews" rather than a huge iFrame for everything is independent from whether it should be using the `EditorContentSlotFill` or not.

Reply:
From memory the main reason we went with a single iframe for everything was in order to correctly show the theme background, whether it's a plain color or a background image. It's particularly relevant for background images as it means the image won't be split up and contained inside multiple small frames, but will replicate what it looks like on the actual site:

<img width="1169" alt="Screenshot 2024-12-06 at 10 14 16 am" src="https://github.com/user-attachments/assets/2071d3dc-d8e3-4bb5-9abc-51c488c9a1d7">

Reply:
Ah that's a good reason :) I wonder if we can have a background iframe positioned absolutely behind everything. (Anyway, I guess we can probably move this discussion to a separate thread/issue :p )